compiler/rustc_target/src/callconv/riscv.rs

```diff
@@ -14,16 +14,18 @@ use crate::spec::HasTargetSpec;
 
 #[derive(Copy, Clone)]
 enum RegPassKind {
-    Float(Reg),
-    Integer(Reg),
+    /// (offset_from_start, type)
+    Float(Size, Reg),
+    /// (offset_from_start, type)
+    Integer(Size, Reg),
     Unknown,
 }
 
 #[derive(Copy, Clone)]
 enum FloatConv {
-    FloatPair(Reg, Reg),
+    FloatPair(Reg, Size, Reg),
     Float(Reg),
-    MixedPair(Reg, Reg),
+    MixedPair(Reg, Size, Reg),
```

Comment:
Is there a meaning to the sequence of these fields? Should it also be recorded similarly using variant fields?